core/src/data/value/error.rs

```diff
@@ -107,6 +110,9 @@ pub enum ValueError {
     #[error("unreachable literal cast from number to float: {0}")]
     UnreachableLiteralCastFromNumberToFloat(String),
 
+    #[error("unreachable literal cast from number to decimal: {0}")]
+    UnreachableLiteralCastFromNumberToDecimal(String),
```

Comment:
There was something we missed.
When converting BigDecimal -> Decimal we are using `FromStr` function.
The following implementation: <https://github.com/paupino/rust-decimal/blob/2b582d625592f1170fc3cc7ee03231bcdce2dd70/src/decimal.rs#L1859-L1866>

The error you're getting here is not unreachable code.
Possible error: <https://github.com/paupino/rust-decimal/blob/2b582d625592f1170fc3cc7ee03231bcdce2dd70/src/error.rs#L6-L14>

Perhaps this should be implemented as an error about conversion...
cc. @panarch

Reply:
Where are we converting from BigDecimal to Decimal?  Or are you referring to the from* function in general?